starlark/testdata/time.star

```diff
@@ -0,0 +1,119 @@
+
+# duration == duration = boolean
+# duration != duration = boolean
+assert.eq(time.parse_duration("1h"), time.parse_duration("1h"))
+assert.ne(time.parse_duration("1h"), time.parse_duration("1m"))
+# duration < duration = boolean
+assert.lt(time.parse_duration("1m"), time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") < time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") < time.parse_duration("1m"))
+# duration <= duration = boolean
+assert.true(time.parse_duration("1m") <= time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") <= time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") <= time.parse_duration("1m"))
+# duration > duration = boolean
+assert.true(not time.parse_duration("1m") > time.parse_duration("1h"))
+assert.true(not time.parse_duration("1h") > time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") > time.parse_duration("1m"))
+# duration >= duration = boolean
+assert.true(not time.parse_duration("1m") >= time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") >= time.parse_duration("1h"))
+assert.true(time.parse_duration("1h") >= time.parse_duration("1m"))
+
+# duration + duration = duration
+assert.eq(d2 + d, time.parse_duration("10h01s"))
+# duration + time = time
+assert.eq(d2 + time.parse_time("2011-04-22T13:33:48Z"), time.parse_time("2011-04-22T23:33:48Z"))
+# duration - duration = duration
+assert.eq(d2 - d, time.parse_duration("9h59m59s"))
+# duration / duration = float
+assert.eq(d2 / time.parse_duration("16m"), 37.5)
+assert.fails(lambda: d2 / time.parse_duration("0"), "division by zero")
+# duration / int = duration
+assert.eq(d2 / 20, time.parse_duration("30m"))
+assert.fails(lambda: d2 / 0, "division by zero")
+# duration // duration = int
+assert.eq(d2 // time.parse_duration("16m"), 37)
+assert.fails(lambda: d2 // time.parse_duration("0"), "division by zero")
+# duration * int = duration
+assert.eq(d * 1000, time.parse_duration("16m40s"))
+assert.fails(lambda: d2 // 0, "division by zero")
+
+
+before = time.now()
+time.sleep(10 * time.millisecond)
+assert.true((time.now() - before).nanoseconds >= time.parse_duration("10ms").nanoseconds)
+
+# time(year=..., month=..., day=..., hour=..., minute=..., second=..., nanosecond=..., location=...)
+t1 = time.time(2009, 6, 12, 12, 6, 10, 99, "US/Eastern")
+assert.eq(t1, time.parse_time("2009-06-12T12:06:10.000000099", format="2006-01-02T15:04:05.999999999", location="US/Eastern"))
+assert.eq(time.time(year=2012, month=12, day=31), time.parse_time("2012-12-31T00:00:00Z"))
+
+# time attributes
+assert.eq(2009, t1.year)
+assert.eq(6, t1.month)
+assert.eq(12, t1.day)
+assert.eq(12, t1.hour)
+assert.eq(6, t1.minute)
+assert.eq(10, t1.second)
+assert.eq(99, t1.nanosecond)
+assert.eq(1244822770, t1.unix)
+assert.eq(1244822770000000099, t1.unix_nano)
+
+# time == time = boolean
+# time != time = boolean
+assert.eq(time.parse_time("2011-04-22T13:33:48Z"), time.parse_time("2011-04-22T13:33:48Z"))
```

Comment:
Define some time and duration constants, and refer to them in the code below; it will make the block easier to read.

e.g. d10h = time.parse_duration("10h")